### PR TITLE
document linearRegression

### PR DESCRIPTION
**Pull request by Fil**
for #1343 

_todo_
- [x] I couldn't make sense of the **reduce** option, PTAL?

### CODE REVIEW

File: src/marks/linearRegression.d.ts [hunk 1]
```diff
@@ -2,23 +2,85 @@ import type {ChannelValue, ChannelValueDenseBinSpec, ChannelValueSpec} from "../
...
+   */
   ci?: number;
+  /**
+   * The distance (in pixels) between samples of the confidence band; defaults to 4.
+   */
   precision?: number;
+  /**
+   * 🌶 How is this used?
+   */
   reduce?: BinReducer;
```
Comment (mbostock) [inline src/marks/linearRegression.d.ts:30]: It’s from maybeDenseInterval. It’s the same idea as line and area. For example for areaY you can specify an interval along _x_, and then reduce defaults to _first_ to extract the first value in _y_ for the current interval in _x_.

File: src/marks/linearRegression.d.ts [hunk 2]
```diff
@@ -2,23 +2,85 @@ import type {ChannelValue, ChannelValueDenseBinSpec, ChannelValueSpec} from "../
...
 
+/** Options for the linearRegressionX and linearRegressionY marks. */
 interface LinearRegressionOptions extends MarkOptions, BinOptions {
+  /**
+   * A channel for the dependent variable **x**.
+   */
   x?: ChannelValueSpec;
+  /**
+   * A channel for the dependent variable **y**.
+   */
```
Comment (Fil) [inline src/marks/linearRegression.d.ts:13]: these represent the dependent variable, and should probably go into their respective mark (x in linearRegressionX, and y in linearRegressionY)

File: src/marks/linearRegression.d.ts (hunk 1 above)
Comment (Fil) [reply src/marks/linearRegression.d.ts]: I've documented what works. Can you open an issue with what doesn't? (It's not clear to me what's missing here.)

File: src/marks/linearRegression.d.ts (hunk 2 above)
Comment (Fil) [reply src/marks/linearRegression.d.ts]: done

**Review by mbostock — APPROVED** [review]

**Pull request merged**